lab_1_classify_by_unigrams/main.py

```diff
@@ -88,20 +183,60 @@ def collect_profiles(paths_to_profiles: list) -> list[dict[str, str | dict[str,
     :paths_to_profiles: a list of strings to the profiles
     :return: a list of loaded profiles
     """
+    if not isinstance(paths_to_profiles, list):
+        return None
+    if not all(isinstance(path, str) for path in paths_to_profiles):
+        return None
+
+    preprocess_profiles_list = []
+    for path in paths_to_profiles:
+        profile = load_profile(path)
+        if isinstance(profile, dict):
+            preprocessed_profile = preprocess_profile(profile)
+            if isinstance(preprocessed_profile, dict):
+                preprocess_profiles_list.append(preprocessed_profile)
 
+    if isinstance(preprocess_profiles_list, list):
```

Comment:
looks strange... I guess you should revork conditions in the loop and return None in bad cases

Reply:
Done